src/main/java/com/commercetools/sync/services/ServiceUtils.java

```diff
@@ -0,0 +1,42 @@
+package com.commercetools.sync.services;
+
+import com.commercetools.sync.commons.BaseSyncOptions;
+import com.commercetools.sync.commons.utils.TriFunction;
+import io.sphere.sdk.commands.DraftBasedCreateCommand;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+public class ServiceUtils {
+
+    /**
+     * Applies the BeforeCreateCallback function on the supplied draft, then attempts to create it on CTP if it's not
+     * empty, then applies the supplied handler on the response from CTP. If the draft was empty after applying the
+     * callback an empty optional is returned as the resulting future.
+     *
+     * @param resourceDraft         draft to apply callback on and then create on CTP.
+     * @param syncOptions           the sync options in which the CTP client is defined on and the callback.
+     * @param createCommandFunction the create command query to create the resource on CTP.
+     * @param responseHandler       defines how to handle the response from CTP.
+     * @param <T>                   the type of the created resource.
+     * @param <S>                   the type of the draft resource to create.
+     * @return a future containing an optional which might contain the resource if successfully creater or empty
+     *         otherwise.
+     */
+    public static <T, S> CompletionStage<Optional<T>> applyCallbackAndCreate(
```

Comment:
why this can't be done inside `syncOptions`? Why we have so many static classes now, which accept `syncOptions`, and then call it's methods wrapping/unwrapping them to Optional. This reminds me old fashion _Procedure Oriented Programming_, like it was in C ages ago.....

Reply:
Much better: 630e995 seems that I got used to static utils.